core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java

```diff
@@ -145,8 +204,141 @@ public int deleteRoleMetasByLegacyTimeLine(long legacyTimeLine, int limit) {
                 SessionUtils.doWithoutCommitAndFetchResult(
                     GroupRoleRelMapper.class,
                     mapper ->
-                        mapper.deleteGroupRoleRelMetasByLegacyTimeLine(legacyTimeLine, limit)));
+                        mapper.deleteGroupRoleRelMetasByLegacyTimeLine(legacyTimeLine, limit)),
+        () ->
+            securableObjectsCount[0] =
+                SessionUtils.doWithoutCommitAndFetchResult(
+                    SecurableObjectMapper.class,
+                    mapper ->
+                        mapper.deleteSecurableObjectsByLegacyTimeLine(legacyTimeLine, limit)));
+
+    return roleDeletedCount[0]
+        + userRoleRelDeletedCount[0]
+        + groupRoleRelDeletedCount[0]
+        + securableObjectsCount[0];
+  }
+
+  long getSecurableObjectEntityId(long metalakeId, String fullName, MetadataObject.Type type) {
+    if (fullName.equals(ALL_METALAKES) && type == MetadataObject.Type.METALAKE) {
+      return 0;
```

Comment:
Never write a hardcode number here like this, it's a bad code, you should define this clearly in some places everyone can know and leverage this.

Reply:
Done.